projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-selection.spec.ts

```diff
@@ -0,0 +1,584 @@
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxGridCellComponent).toBe(true);
+            expect(treeGrid.selectedCells[0].value).toBe(475);
+
+            // level 3
+            treeGridCell = TreeGridFunctions.getTreeCell(rows[2]);
+            treeGridCell.triggerEventHandler('focus', new Event('focus'));
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxGridCellComponent).toBe(true);
+            expect(treeGrid.selectedCells[0].value).toBe(957);
+        });
+
+        it('should persist selection after paging', () => {
+            const rows = TreeGridFunctions.getAllRows(fix);
+            const treeGridCell = TreeGridFunctions.getTreeCell(rows[0]);
+            treeGridCell.triggerEventHandler('focus', new Event('focus'));
+            fix.detectChanges();
+
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxTreeGridCellComponent).toBe(true);
+            expect(TreeGridFunctions.verifyGridCellHasSelectedClass(treeGridCell)).toBe(true);
+
+            navigateToNextPage(fix);
+            navigateToFirstPage(fix);
+            fix.detectChanges();
+
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxTreeGridCellComponent).toBe(true);
+            expect(TreeGridFunctions.verifyGridCellHasSelectedClass(treeGridCell)).toBe(true);
+
+            navigateToLastPage(fix);
+            navigateToFirstPage(fix);
+            fix.detectChanges();
+
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxTreeGridCellComponent).toBe(true);
+            expect(TreeGridFunctions.verifyGridCellHasSelectedClass(treeGridCell)).toBe(true);
+        });
+
+        it('should persist selection after filtering', () => {
+            const rows = TreeGridFunctions.getAllRows(fix);
+            const treeGridCell = TreeGridFunctions.getTreeCell(rows[0]);
+            treeGridCell.triggerEventHandler('focus', new Event('focus'));
+
+            treeGrid.filter('ID', '14', IgxStringFilteringOperand.instance().condition('startsWith'), true);
+            fix.detectChanges();
+
+            expect(treeGrid.selectedCells.length).toBe(1);
+            expect(treeGrid.selectedCells[0] instanceof IgxTreeGridCellComponent).toBe(true);
+            expect(TreeGridFunctions.verifyGridCellHasSelectedClass(treeGridCell)).toBe(true);
+
+            // set new filtering
+            treeGrid.clearFilter('ProductName');
+            treeGrid.filter('ID', '8', IgxStringFilteringOperand.instance().condition('startsWith'), true);
+            fix.detectChanges();
+
+            expect(treeGrid.selectedCells.length).toBe(0);
+        });
+
+        it('should persist selection after scrolling', async () => {
```

Comment:
Make sure tests involving scrolling are async because of the requestAnimationFrame usage. (Already fixed in this branch)

Reply:
ok